src/patternfly/components/FormControls/styles.scss

```diff
@@ -34,24 +60,38 @@
   &::placeholder {
     color: var(--pf-c-form-control--placeholder--Color);
   }
-
+  &.pf-m-disabled,
   &:disabled {
     color: var(--pf-c-form-control--disabled--Color);
     cursor: not-allowed;
     background-color: var(--pf-c-form-control--disabled--BackgroundColor);
     border-color: var(--pf-c-form-control--disabled--BorderColor);
   }
+  &.pf-m-error {
```

Comment:
I think the error class needs to be `.pf-m-invalid` when applied to an input, as modifiers typically match the pseudo classes they mimic. Alternatively, you could do away with `.pf-m-invalid` and just use `required`. So the styling would look more like this example:

This example would apply error styling to a valid, required field
```
&.pf-m-error,
[attr="required"] {
  border-color: var(--pf-c-form-control--error--BorderColor);
  border-width: var(--pf-c-form-control--error--BorderSize);
}
```

This would only apply to an invalid input
```
&.pf-m-invalid,
&:invalid {
  border-color: var(--pf-c-form-control--error--BorderColor);
  border-width: var(--pf-c-form-control--error--BorderSize);
}
```
This would only apply when input becomes valid
```
&.pf-m-valid,
&:valid {
  border-color: green;
}
```
Additionally, `valid` inputs can be enhanced when they become valid, like showing a green check and border, for example.

`.pf-m-error` would still be applied to helper text.

More on topic: https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation#Using_built-in_form_validation

Reply:
So just to make sure I'm clear on what you're saying: Change 'pf-m-error' to 'pf-m-invalid' and add the pseudo class? I think that makes sense. 

The only styles we have for a required field are on the label. I don't believe we are including styles for valid fields. 

I remember a conversation between @mcarrano and @mceledonia that dropped them.

Reply:
@dmiller9911 the modifiers will remain where attributes don't exist. So in the case of the inputs we would remove the `.pf-m-disabled` modifier and style just the attribute selector.